agent/agent_endpoint.go

```diff
@@ -457,6 +457,91 @@ func (s *HTTPServer) AgentCheckUpdate(resp http.ResponseWriter, req *http.Reques
 	return nil, nil
 }
 
+func AgentHealthService(serviceId string, s *HTTPServer, resp http.ResponseWriter, req *http.Request) (int, string) {
+	checks := s.agent.State.Checks()
+	serviceChecks := make(api.HealthChecks, 0)
+	for _, c := range checks {
+		if c.ServiceID == serviceId || c.ServiceID == "" {
+			// TODO: harmonize struct.HealthCheck and api.HealthCheck (or at least extract conversion function)
+			healthCheck := &api.HealthCheck{
+				Node:        c.Node,
+				CheckID:     string(c.CheckID),
+				Name:        c.Name,
+				Status:      c.Status,
+				Notes:       c.Notes,
+				Output:      c.Output,
+				ServiceID:   c.ServiceID,
+				ServiceName: c.ServiceName,
+				ServiceTags: c.ServiceTags,
+			}
+			serviceChecks = append(serviceChecks, healthCheck)
+		}
+	}
+	status := serviceChecks.AggregatedStatus()
+	switch status {
+	case api.HealthWarning:
+		return http.StatusTooManyRequests, status
+	case api.HealthPassing:
+		return http.StatusOK, status
+	default:
+		return http.StatusServiceUnavailable, status
+	}
+}
+
+func (s *HTTPServer) AgentHealthServiceId(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+
+	// Pull out the service id (service id since there may be several instance of the same service on this host)
+	serviceID := strings.TrimPrefix(req.URL.Path, "/v1/agent/health/service/id/")
+	if serviceID == "" {
+		resp.WriteHeader(http.StatusBadRequest)
```

Comment:
I agree, but I just mimicked what was already present in other endpoints, for instance : `AgentServiceMaintenance` which does exactly the same kind of checks.

Returning the error from this function means it would be probably handled and logged by Consul as an error right?

I can change it, but I am not sure of the semantic differences it implies

Reply:
Ignore this for now. I talked it over with @banks and I think we will probably do a larger refactor of our http error handling later including propagating error types out to the main http code instead of a piecemeal approach.